src/main/java/io/aiven/kafka/connect/transforms/FilterFields.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2023 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.connect.transforms;
+
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.kafka.common.config.AbstractConfig;
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.connect.connector.ConnectRecord;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.transforms.Transformation;
+
+public class FilterFields<R extends ConnectRecord<R>> implements Transformation<R> {
+
+    private String fieldName;
+    private String fieldValue;
+    private Pattern fieldValuePattern;
+    private String filterMode;
+
+    public FilterFields() {
+        this.fieldName = "";
+        this.fieldValue = "";
+        this.fieldValuePattern = Pattern.compile("");
+        this.filterMode = "equals";
+    }
+
+    @Override
+    public R apply(final R record) {
+
+        if (!(record.value() instanceof Struct)) {
+            return record;
+        }
+
+        final Struct value = (Struct) record.value();
+        final Object fieldValueObj = value.get(fieldName);
+
+        if (fieldValueObj != null) {
+            final Matcher matcher = fieldValuePattern.matcher(fieldValueObj.toString());
+            final boolean matches = matcher.matches();
+
+            if (("equals".equalsIgnoreCase(filterMode) && matches)
+                    || ("not_equals".equalsIgnoreCase(filterMode) && !matches)) {
```

Comment:
May move the config validation to the constructor? Also, with the config value change proposed, should be nicer to compare

Reply:
moved to configure()